prow/cmd/tide/README.md

```diff
@@ -24,7 +24,96 @@ them when they have up-to-date passing test results ("tide goes out").
+
+## Flowchart
+
+```mermaid
+    graph TD;
+        subgraph github[GitHub]
+            subgraph org/repo/branch
+                head-ref[HEAD ref];
+                pullrequest[Pull Request];
+                status-context[Status Context];
+            end
+        end
+
+        subgraph prow-cluster
+            prowjobs[Prowjobs];
+            config.yaml;
+        end
+
+        subgraph tide-workflow
+            Tide;
+            pools;
+            filtered-pools
+            subgraph sync-sub-pool
+                single-pool;
+                pool-n;
+                pool-n1;
+                accumulated-batch-prowjobs-->|filter out <br> incorrect refs <br> no longer meet merge requirement|valid-batches;
+                valid-batches-->accumulated-batch-success;
+                valid-batches-->accumulated-batch-pending;
+                
+                status-context-->filtered-prowjobs;
+                filtered-prowjobs-->|accumulate|map_context_best-result;
+                map_context_best-result-->map_pr_overall-results;
+                map_pr_overall-results-->accumulated-success;
+                map_pr_overall-results-->accumulated-pending;
+                map_pr_overall-results-->accumulated-stale;
+                
+                subgraph all-accumulated-pools
+                    accumulated-batch-success;
+                    accumulated-batch-pending;
+                    accumulated-success;
+                    accumulated-pending;
+                    accumulated-stale;
+                end
+
+                accumulated-batch-success-..->accumulated-batch-success-exist{Exist};
+                accumulated-batch-pending-..->accumulated-batch-pending-exist{Exist};
+                accumulated-success-..->accumulated-success-exist{Exist};
+                accumulated-pending-..->accumulated-pending-exist{Exist};
+                accumulated-stale-..->accumulated-stale-exist{Exist};
+                accumulated-batch-success-exist-->|yes|merge-batch[Merge batch];
+                accumulated-batch-success-exist-->|no|accumulated-batch-pending-exist;
+                accumulated-batch-pending-exist-->|no|accumulated-success-exist;
+                accumulated-success-exist-->|yes|merge-single[Merge Single];
+                accumulated-success-exist-->|no|accumulated-pending-exist;
+                accumulated-pending-exist-->|no|can-trigger-batch{Can Trigger New Batch};
+                can-trigger-batch-->|yes|trigger-batch[Trigger new batch];
+                can-trigger-batch-->|no|accumulated-stale-exist;
+                accumulated-stale-exist-->|yes|trigger-highest-pr[Trigger Jobs on Highest Priority PR];
+            end
+        end
+
+        Tide-->pools;
+        pullrequest-->pools[Pools - grouped PRs, prow jobs by org/repo/branch];
```

Comment:
Pools are actually just the PRs by org/repo/number, only after that in `divicePool` we get the Prowjobs and construct unfiltered subpoools

Reply:
That is the divide pool step, perhaps we should label it as such.

Reply:
will need to think about how to do this better

Reply:
done